src/Scripts/UI/Controls/GraphicsEditor.gd

```diff
@@ -195,96 +257,551 @@ func image_draw(target_image: Image, pos: Vector2, color: Color, point_size: int
+		if active_transfer_button:
+			# Check if active_transfer_button is still a valid node
+			if is_instance_valid(active_transfer_button): 
+				active_transfer_button.modulate = Color.WHITE
+			active_transfer_button = null
+
+		# Activate the new button
+		active_transfer_button = rotate_button
+		active_transfer_button.modulate = Color.LIME_GREEN
+		_rotating = true
+		_rotation_pivot = _layers_container.get_local_mouse_position()
+
+func _scale(Hbox: HBoxContainer) -> void:
+	var hbox_index = %LayersList.get_children().find(Hbox)
+	var scale_button = Hbox.get_child(4) 
+
+	# Toggle visibility of child nodes in the layer FIRST
+	var layer = _layers_container.get_child(hbox_index)
+	for child in layer.get_children():
+		child.visible = !child.visible 
+
+	# --- Toggle Logic (Fixed) ---
+	if scale_button == active_transfer_button:
+		# Deactivate if the same button is pressed again
+		active_transfer_button.modulate = Color.WHITE
+		active_transfer_button = null 
+	else:
+		# Deactivate the previous button
+		if active_transfer_button:
+			if is_instance_valid(active_transfer_button):
+				active_transfer_button.modulate = Color.WHITE
+			active_transfer_button = null
+
+		# Activate the new button
+		active_transfer_button = scale_button
+		active_transfer_button.modulate = Color.LIME_GREEN 
+		
+func _on_arrowleft_pressed() -> void:
+	_resize_layers(1.1, 1.0)  # Increase width by 10%, center horizontallyc
 	
+	# --- Move layers after resizing ---
+	for layer in _layers_container.get_children():
+		if layer is Layer:
+			layer.position.x -= layer.size.x * 0.09 # Move right by 5% of the new width 
+			
+func _on_arrow_right_pressed() -> void:
+	_resize_layers(1.1, 1.0)  # Increase width by 10%, center horizontally
+
+func _on_arrow_top_pressed() -> void:
+	_resize_layers(1.1,false) # Increase height by 10%, center vertically 
+	# --- Move layers after resizing ---
+	for layer in _layers_container.get_children():
+		if layer is Layer:
+			layer.position.y -= layer.size.y * 0.09 # Move right by 5% of the new width 
+
+func _on_arrow_bottom_pressed() -> void:
+	_resize_layers(1.1,false) # Increase height by 10%, center vertically 
+
+func _resize_layers(size_factor: float, resize_width: bool = true) -> void:
+	for layer in _layers_container.get_children():
+		if layer is Layer:
+			var old_size = layer.image.get_size()
+			var new_size: Vector2
+
+			if resize_width:
+				new_size = Vector2i(old_size.x * size_factor, old_size.y)
+			else:
+				new_size = Vector2i(old_size.x, old_size.y * size_factor)
+
+			# Store the original content of the layer
+			var temp_image := Image.new()
+			temp_image.copy_from(layer.image)
+			# Resize the layer's image
+			layer.image.resize(new_size.x, new_size.y, Image.INTERPOLATE_BILINEAR)
+			layer.size = new_size
+
+			# Redraw the original content onto the resized image
+			layer.image.blit_rect(temp_image, Rect2(Vector2.ZERO, old_size), Vector2.ZERO)
+			layer.update()
+
+func layers_buttons():
```

Comment:
we seem to be adding a set of buttons for every layer that exists, it's better if we create only one set that works for all the layers ( this can be done by emitting a signal and checking the active layer in the method that catches the signal and then doing what the button is supposed to do)